python/pyarrow/tests/test_array.py

```diff
@@ -2460,18 +2460,24 @@ def test_buffers_nested():
     assert struct.unpack('4xh', values) == (43,)
 
 
-def test_nbytes_sizeof():
+def test_total_buffer_size():
```

Comment:
Can we add just one test where the array is sliced so that `nbytes` and `get_total_buffer_size` return different values?

Reply:
added a test case.